File: tests/verify/managed/fixtures/clean/clean.ts
// clean.ts — clean TypeScript fixture: ZERO findings expected.
// Proves hot scoping (cold-path allocations below are legal).

/** @hot */
export function onTick(tick: number, out: Float64Array, scratch: Float64Array): void {
  let acc = 0;
  for (let i = 0; i < 8; i++) {
    scratch[i] = tick * i;
    const magnitude = scratch[i] > 0 ? scratch[i] : -scratch[i];
    acc += magnitude;
  }
  out[0] = acc;
  if (acc > 1e9) {
    out[1] = -1;
    return;
  }
}

export function coldPath(tick: number): number {
  const fresh = new Float64Array(8); // NOT hot: legal
  const cb = (x: number) => x + tick; // NOT hot: legal
  return fresh[0] + cb(1);
}

File: tests/verify/managed/fixtures/clean/clean.rs
// clean.rs — clean Rust fixture: ZERO findings expected.
// with_capacity-style preallocation at init is the compliant pattern.

#[weft_hot]
fn on_quote(q: &Quote, out: &mut Out) {
    let mut acc: f64 = 0.0;
    for i in 0..64 {
        acc += q.window[i];
    }
    out.acc = acc;
    let scalar = q.px as f64;
    out.last = scalar * 0.5;
}

fn cold() -> String {
    format!("{:?}", 42) // NOT hot: legal
}
